src/main/java/anodot/stage/destination/AnodotTarget.java

```diff
@@ -139,6 +147,40 @@ public void write(Batch batch) throws StageException {
         }
     }
 
+    private Record getLastRecord(Batch batch) {
+        Iterator<Record> iter = batch.getRecords();
+        Record record = null;
+        while (iter.hasNext()) {
+            record = iter.next();
+        }
+        return record;
+    }
+
+    private void sendOffsetToAgent(String offset, String pipelineId) throws Exception {
+        URLConnection con = new URL(conf.agentOffsetUrl + pipelineId).openConnection();
+        HttpURLConnection http = (HttpURLConnection)con;
+        http.setRequestMethod("POST");
+        http.setDoOutput(true);
+        byte[] out = String.format("{\"offset\": \"%s\"}", offset).getBytes(StandardCharsets.UTF_8);
+        http.setFixedLengthStreamingMode(out.length);
+        http.setRequestProperty("Content-Type", "application/json; charset=UTF-8");
+        http.connect();
+        try (OutputStream os = http.getOutputStream()) {
+            os.write(out);
+        }
+        if (http.getResponseCode() != 200) {
+            try(BufferedReader br = new BufferedReader(
+                    new InputStreamReader(con.getInputStream(), StandardCharsets.UTF_8))) {
+                StringBuilder response = new StringBuilder();
+                String responseLine;
+                while ((responseLine = br.readLine()) != null) {
+                    response.append(responseLine.trim());
+                }
+                throw new Exception(response.toString());
```

Comment:
maybe should be replaced with 

` new OnRecordErrorException()` or something not generic like `Exception`

Reply:
well, yeah, it just wasn't meant to be handled so I put Exception here

Reply:
maybe add more context than?
Something like `failed to saved agent timestamp. HTTP error ${response_code}: ${response_body}`